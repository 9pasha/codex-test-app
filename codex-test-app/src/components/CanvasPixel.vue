<template>
  <div class="canvas-px">{{ color }}</div>
</template>

<script>
export default {
  name: "CanvasPixel",
  props: {
    x: Number,
    y: Number
  },
  computed: {
    color() {
      return this.$store.state.pixels[this.y - 1][this.x - 1];
    }
  }
};
</script>

<style scoped>
.canvas-px {
  width: 45px;
  height: 45px;
  border: 1px solid dodgerblue;
  display: flex;
  align-items: center;
  justify-content: center;
}
</style>
