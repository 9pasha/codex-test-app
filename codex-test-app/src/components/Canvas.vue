<template>
  <div class="canvas">
    <div class="row no-gutters flex-nowrap justify-center"
    v-for="rowItm in row"
    :key="rowItm">
      <div class="canvas-col"
      v-for="colItm in col"
      :key="colItm">
        <canvas-pixel></canvas-pixel>
      </div>
    </div>
  </div>
</template>

<script>
import CanvasPixel from "@/components/CanvasPixel";
export default {
  name: "CanvasField",
  components: {
    CanvasPixel,
  },
  data: () => ({
  }),
  computed: {
    row() {
      return this.$store.state.canvas.y;
    },
    col() {
      return this.$store.state.canvas.x;
    },
  },
};
</script>

<style scoped>
.canvas {
  width: 100%;
  overflow: auto;

  &-col {
    width: 40px;
    height: 40px;
  }
}
</style>
