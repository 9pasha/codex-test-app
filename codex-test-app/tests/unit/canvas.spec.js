import { shallowMount, createLocalVue } from "@vue/test-utils";
import Canvas from "@/components/Canvas.vue";
import CanvasPixel from "@/components/CanvasPixel";
import Vuex from "vuex";

const localVue = createLocalVue();

localVue.use(Vuex);

describe("Canvas.vue", () => {
  let store;

  beforeEach(() => {
    store = new Vuex.Store({
      state: {
        canvas: {
          x: 3,
          y: 10
        },
        isCorrectCanvas: false
      }
    });
  });
  it("check error alert", () => {
    const wrapper = shallowMount(Canvas, {
      localVue,
      store
    });
    const errorMsg =
      "Sorry, canvas can't be created. At first line of input.txt you need to create canvas(C x y) and set min-size: 2x2.";

    expect(wrapper.is(Canvas)).toBeTruthy();
    expect(wrapper.contains(CanvasPixel)).toBe(false);
    const errorEl = wrapper.find("h2");
    expect(errorEl.text()).toEqual(errorMsg);
  });
  it("check render without error", () => {
    store.state.isCorrectCanvas = true;
    const wrapper = shallowMount(Canvas, {
      localVue,
      store
    });

    expect(wrapper.is(Canvas)).toBeTruthy();
    expect(wrapper.findComponent(CanvasPixel)).toBeTruthy();
    const pixels = wrapper.findAllComponents(CanvasPixel);
    expect(pixels).toHaveLength(30);
    expect(wrapper.contains(".canvas h2")).toBe(false);
  });
  it("should match snapshots", () => {
    store.state.isCorrectCanvas = true;
    const wrapper = shallowMount(Canvas, {
      localVue,
      store
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
