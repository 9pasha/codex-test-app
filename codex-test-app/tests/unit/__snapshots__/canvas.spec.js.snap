// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Canvas.vue should match snapshots 1`] = `
<div class="canvas">
  <div class="wrapper">
    <div class="row no-gutters flex-nowrap justify-center">
      <div class="canvas-col">
        <canvas-pixel-stub x="1" y="1"></canvas-pixel-stub>
      </div>
      <div class="canvas-col">
        <canvas-pixel-stub x="2" y="1"></canvas-pixel-stub>
      </div>
      <div class="canvas-col">
        <canvas-pixel-stub x="3" y="1"></canvas-pixel-stub>
      </div>
    </div>
    <div class="row no-gutters flex-nowrap justify-center">
      <div class="canvas-col">
        <canvas-pixel-stub x="1" y="2"></canvas-pixel-stub>
      </div>
      <div class="canvas-col">
        <canvas-pixel-stub x="2" y="2"></canvas-pixel-stub>
      </div>
      <div class="canvas-col">
        <canvas-pixel-stub x="3" y="2"></canvas-pixel-stub>
      </div>
    </div>
    <div class="row no-gutters flex-nowrap justify-center">
      <div class="canvas-col">
        <canvas-pixel-stub x="1" y="3"></canvas-pixel-stub>
      </div>
      <div class="canvas-col">
        <canvas-pixel-stub x="2" y="3"></canvas-pixel-stub>
      </div>
      <div class="canvas-col">
        <canvas-pixel-stub x="3" y="3"></canvas-pixel-stub>
      </div>
    </div>
    <div class="row no-gutters flex-nowrap justify-center">
      <div class="canvas-col">
        <canvas-pixel-stub x="1" y="4"></canvas-pixel-stub>
      </div>
      <div class="canvas-col">
        <canvas-pixel-stub x="2" y="4"></canvas-pixel-stub>
      </div>
      <div class="canvas-col">
        <canvas-pixel-stub x="3" y="4"></canvas-pixel-stub>
      </div>
    </div>
    <div class="row no-gutters flex-nowrap justify-center">
      <div class="canvas-col">
        <canvas-pixel-stub x="1" y="5"></canvas-pixel-stub>
      </div>
      <div class="canvas-col">
        <canvas-pixel-stub x="2" y="5"></canvas-pixel-stub>
      </div>
      <div class="canvas-col">
        <canvas-pixel-stub x="3" y="5"></canvas-pixel-stub>
      </div>
    </div>
    <div class="row no-gutters flex-nowrap justify-center">
      <div class="canvas-col">
        <canvas-pixel-stub x="1" y="6"></canvas-pixel-stub>
      </div>
      <div class="canvas-col">
        <canvas-pixel-stub x="2" y="6"></canvas-pixel-stub>
      </div>
      <div class="canvas-col">
        <canvas-pixel-stub x="3" y="6"></canvas-pixel-stub>
      </div>
    </div>
    <div class="row no-gutters flex-nowrap justify-center">
      <div class="canvas-col">
        <canvas-pixel-stub x="1" y="7"></canvas-pixel-stub>
      </div>
      <div class="canvas-col">
        <canvas-pixel-stub x="2" y="7"></canvas-pixel-stub>
      </div>
      <div class="canvas-col">
        <canvas-pixel-stub x="3" y="7"></canvas-pixel-stub>
      </div>
    </div>
    <div class="row no-gutters flex-nowrap justify-center">
      <div class="canvas-col">
        <canvas-pixel-stub x="1" y="8"></canvas-pixel-stub>
      </div>
      <div class="canvas-col">
        <canvas-pixel-stub x="2" y="8"></canvas-pixel-stub>
      </div>
      <div class="canvas-col">
        <canvas-pixel-stub x="3" y="8"></canvas-pixel-stub>
      </div>
    </div>
    <div class="row no-gutters flex-nowrap justify-center">
      <div class="canvas-col">
        <canvas-pixel-stub x="1" y="9"></canvas-pixel-stub>
      </div>
      <div class="canvas-col">
        <canvas-pixel-stub x="2" y="9"></canvas-pixel-stub>
      </div>
      <div class="canvas-col">
        <canvas-pixel-stub x="3" y="9"></canvas-pixel-stub>
      </div>
    </div>
    <div class="row no-gutters flex-nowrap justify-center">
      <div class="canvas-col">
        <canvas-pixel-stub x="1" y="10"></canvas-pixel-stub>
      </div>
      <div class="canvas-col">
        <canvas-pixel-stub x="2" y="10"></canvas-pixel-stub>
      </div>
      <div class="canvas-col">
        <canvas-pixel-stub x="3" y="10"></canvas-pixel-stub>
      </div>
    </div>
  </div>
  <!---->
</div>
`;
